### PR TITLE
ccbuilder integration

### PR DESCRIPTION
**Pull request by Quarub**
Depends on  https://github.com/DeadCodeProductions/ccbuilder/pull/1 and  https://github.com/DeadCodeProductions/ccbuilder/pull/2

Checks fail because ccbuilder v0.0.4 is not yet released.

### CODE REVIEW

**Comment by thetheodor** [comment]
[a6bd48d](https://github.com/DeadCodeProductions/dead/pull/20/commits/a6bd48d0aac984dc58a14612abf18c11f122af00) looks similar to [0f01035](https://github.com/DeadCodeProductions/dead/commit/0f01035cf2c0b565a05e8cde61a83dee09b54212)

**Comment by Quarub** [comment]
[a6bd48d](https://github.com/DeadCodeProductions/dead/pull/20/commits/a6bd48d0aac984dc58a14612abf18c11f122af00) in general looks horrible.
It seems I messed up when rebasing.

File: checker.py [hunk 1]
```diff
@@ -361,25 +363,27 @@ def is_interesting_with_static_globals(self, case: utils.Case) -> bool:
...
+                asm_good = utils.get_asm_str(static_code, good_setting, self.builder)
                 if case.marker in asm_good:
                     uninteresting = True
                     break
             return not uninteresting
 
     def _emtpy_marker_code_str(self, case: utils.Case) -> str:
         marker_prefix = utils.get_marker_prefix(case.marker)
-        p = re.compile(f"void {marker_prefix}(.*)\(void\);")
+        p = re.compile(f"void {marker_prefix}(.*)\(void\);(.*)")
```
Comment (thetheodor) [inline checker.py:379]: This should be `fr"void {marker_prefix}(.*)\(void\);(.*)"`

File: checker.py [hunk 2]
```diff
@@ -361,25 +363,27 @@ def is_interesting_with_static_globals(self, case: utils.Case) -> bool:
...
         empty_body_code = ""
         for line in case.code.split("\n"):
             m = p.match(line)
             if m:
-                empty_body_code += f"\nvoid {marker_prefix}{m.group(1)}(void){{}}"
+                empty_body_code += (
+                    f"\nvoid {marker_prefix}{m.group(1)}(void){{}}\n{m.group(2)}"
```
Comment (thetheodor) [inline checker.py:385]: `fr"\nvoid {marker_prefix}{m.group(1)}(void){{}}\n{m.group(2)}"`

File: utils.py [hunk 3]
```diff
@@ -504,8 +511,8 @@ def run_cmd(
         cmd, cwd=str(working_dir), check=True, env=env, capture_output=True, **kwargs
     )
 
-    logging.debug(output.stdout.decode("utf-8").strip())
-    logging.debug(output.stderr.decode("utf-8").strip())
+    # logging.debug(output.stdout.decode("utf-8").strip())
+    # logging.debug(output.stderr.decode("utf-8").strip())
```
Comment (thetheodor) [inline utils.py:515]: Drop?

**Review by thetheodor — CHANGES_REQUESTED** [review]

**Review by thetheodor — APPROVED** [review]

**Pull request merged**